### PR TITLE
new way to call bslib

### LINKED ISSUE #1695: Should bslib usage happen only once
At the moment, the way `data_deps()` is called in `render_page()` mean bslib is called for every page which is extremely wasteful as we do not allow bslib variables in the YAML.

So instead, should `data_deps()` be memoised? Or should there be more refactoring? `data_deps()` depends on `depth`.

### PR DESCRIPTION
**Pull request by maelle**
Fix #1695 

Other approaches (thanks @cderv for the insightful discussion) would involve still calling bslib & friends in `render_page()`, but **caching** the result somehow (maybe "stateful function", maybe `rlang::env_cache()` -- again thanks @cderv).

Pros of the approach of this PR
+ to me it seems more logical to create assets in `init_site()`, when I first thought of caching I had the impression to apply a band-aid on bad design of mine.
+ you can run `init_site()` once and be done with it for several R sessions where you e.g. only work on articles.

Con of the approach of this PR
+ it is a bit awkward to save tags in a text file.
+ if you change the bslib params in the configuration and build articles, you won't see the new styling.

Note that this PR uses regex for now :sweat_smile: but the tags could be tweaked with XPath.

### CODE REVIEW

File: R/render.r [hunk 1]
```diff
@@ -408,93 +408,27 @@ footer_pkgdown <- function(data) {
...
 
-  abort(
-    sprintf(
-      "Can't find Bootswatch theme '%s' (%s) for Bootstrap version '%s' (%s).",
-      bootswatch_theme,
-      pkgdown_field(pkg = pkg, "template", "bootswatch"),
-      bs_version,
-      pkgdown_field(pkg = pkg, "template", "bootstrap")
-    )
+  c(
```
Comment (maelle) [inline R/render.r:424]: Added a `paste0(, collapse = "")` around this in #1740.

**Comment by maelle** [comment]
Not to be merged but this shows the change better than in #1740.

**Comment by hadley** [comment]
What's the status of this PR?

**Comment by maelle** [comment]
These changes are part of #1740

**Pull request closed without merging**